src/basic/terminal-util.c

```diff
@@ -1212,11 +1213,8 @@ bool colors_enabled(void) {
         static int enabled = -1;
 
         if (_unlikely_(enabled < 0)) {
-                const char *colors;
-
-                colors = getenv("SYSTEMD_COLORS");
-                if (colors)
-                        enabled = parse_boolean(colors) != 0;
+                if (getenv_bool("SYSTEMD_COLORS") > 0)
+                        enabled = true;
```

Comment:
i figure setting SYSTEMD_COLORS=0 should really turn off colors...

Hence this should probably be:

```c
b = getenv_bool("SYSTEMD_COLORS");
if (b >= 0)
        enabled = b;
else …
```